### ISSUE TITLE
Display unassigned task on challenge listings page

### ISSUE THREAD

**Issue by Oanh-and-only-Oanh** [topcoder-platform/community-app#5712]
New feature description:

When an unassigned task is created in WM, display this unassigned task on the challenge listings page under the "open for registration" bucket.
-any authenticated member can register for the challenge
-once the ticket has been picked up by a member, the task should be removed from the "open for registration" bucket and appear under "all challenge" 
-member can be unassigned by the copilot or they can unassign themselves from the task
-when member registers from the task, the task should reappear under "open for registration" bucket and be open for registration again

-An unassigned task should be accompanied with a forum discussion

**Comment by dedywahyudi** [comment]
@Oanh-and-only-Oanh Okay, i need ask some developer help to fix all these. I'll share once i received the solution

**Comment by drasticdpk** [comment]
- And challenge details should be display without login to application.
- Currently it is showing in listing page but not on details page. 

Reference 
https://monosnap.com/file/Op50UkkgdZ3E94KOQtIXl81Zf6VsBE
![image](https://user-images.githubusercontent.com/25297859/134625061-481fefc2-c109-41aa-b4cc-8b5916429547.png)


https://community-app.topcoder-dev.com/challenges/d2e69005-ffda-45ee-a70d-cab4d477fae8

cc @SathyaJayabal

**Comment by drasticdpk** [comment]
Dev verification 

**Basic flow**
https://user-images.githubusercontent.com/25297859/135023043-3fe78aa3-a45c-4923-9116-c8f1991d6528.mp4

**Data Science Task with forums off** 

![image](https://user-images.githubusercontent.com/25297859/135022245-e055f9da-cb82-4041-a8d8-f5a63df04c6a.png)


**Design task( Updating forums after unregister the task )**

![image](https://user-images.githubusercontent.com/25297859/135023867-b0c43974-7c66-4125-9100-bb5a67ddd6c8.png)

**Comment by dedywahyudi** [comment]
>     * And challenge details should be display without login to application.
> 
>     * Currently it is showing in listing page but not on details page.
> 
> 
> Reference https://monosnap.com/file/Op50UkkgdZ3E94KOQtIXl81Zf6VsBE ![image](https://user-images.githubusercontent.com/25297859/134625061-481fefc2-c109-41aa-b4cc-8b5916429547.png)
> 
> https://community-app.topcoder-dev.com/challenges/d2e69005-ffda-45ee-a70d-cab4d477fae8
> 
> cc @SathyaJayabal

@drasticdpk so, for this challenge details page need to visible without user need to login?

**Comment by dedywahyudi** [comment]
> Dev verification
> 
> **Basic flow** https://user-images.githubusercontent.com/25297859/135023043-3fe78aa3-a45c-4923-9116-c8f1991d6528.mp4
> 
> **Data Science Task with forums off**
> 
> ![image](https://user-images.githubusercontent.com/25297859/135022245-e055f9da-cb82-4041-a8d8-f5a63df04c6a.png)
> 
> **Design task( Updating forums after unregister the task )**
> 
> ![image](https://user-images.githubusercontent.com/25297859/135023867-b0c43974-7c66-4125-9100-bb5a67ddd6c8.png)

@drasticdpk 
can you explain what need to do with these 2 images?

**Comment by dedywahyudi** [comment]
> New feature description:
> 
> When an unassigned task is created in WM, display this unassigned task on the challenge listings page [https://www.topcoder.com/challenges?tracks[DS]=true&tracks[Des]=true&tracks[Dev]=true&tracks[QA]=true&types[]=TSK](https://www.topcoder.com/challenges?tracks%5BDS%5D=true&tracks%5BDes%5D=true&tracks%5BDev%5D=true&tracks%5BQA%5D=true&types%5B%5D=TSK) under the "open for registration" bucket. -any authenticated member can register for the challenge -once the ticket has been picked up by a member, the task should be removed from the "open for registration" bucket and appear under "all challenge" -member can be unassigned by the copilot or they can unassign themselves from the task -when member registers from the task, the task should reappear under "open for registration" bucket and be open for registration again
> 
> -An unassigned task should be accompanied with a forum discussion

@Oanh-and-only-Oanh 

An unassigned task should be accompanied with a forum discussion
- So, for all unassigned task need automatically create the forum discussion? How about user select Forum Discussion Off on Step 1 of Launch New Challenge? They not overlapping?

**Comment by drasticdpk** [comment]
> @drasticdpk can you explain what need to do with these 2 images?
This is not issue, it is verification reference cc @dedywahyudi

**Comment by drasticdpk** [comment]
> @drasticdpk so, for this challenge details page need to visible without user need to login?
yes correct , challenge should be visible to public without login to application.

**Comment by luizrrodrigues** [comment]
@dedywahyudi this one related with this PR https://github.com/topcoder-platform/community-app/pull/5713 ?

**Comment by dedywahyudi** [comment]
@luizrrodrigues yes, but still not complete yet. 
I'll let you know later

**Comment by luizrrodrigues** [comment]
@dedywahyudi Oh okay, I'll remove from release today then.

**Comment by dedywahyudi** [comment]
@Oanh-and-only-Oanh @drasticdpk @luizrrodrigues 

PR for this ready, please test & review
https://github.com/topcoder-platform/topcoder-react-lib/pull/316
https://github.com/topcoder-platform/community-app/pull/5731

**Comment by luizrrodrigues** [comment]
@SathyaJayabal Ready for QA in Stag and Dev.

cc @dedywahyudi @Oanh-and-only-Oanh

**Comment by luizrrodrigues** [comment]
Deploying to Stag and Dev again.

**Comment by drasticdpk** [comment]
Verified on Dev env. 
**Without user login challenge is listing and register button is enable.**
![image](https://user-images.githubusercontent.com/25297859/136341275-33d7d3ad-730e-459d-99a8-472d8df7b376.png)

**Forums is showing for register user ( with forums enable from WM)**
![image](https://user-images.githubusercontent.com/25297859/136341692-5cc3f335-e5ad-4b6f-a194-b6f6ec5eef80.png)

**Task user has been assign and task is completed** 
![image](https://user-images.githubusercontent.com/25297859/136342068-2a55cbdc-4c89-434f-8ef5-a0235f78ccd1.png)

**Comment by drasticdpk** [comment]
Register button does not enable for Challenge type TASK 
Reference : 
![image](https://user-images.githubusercontent.com/25297859/136384112-f7118f33-8107-44e1-a393-f97352d14b2c.png)
Reference : 
http://www.topcoder.com/challenges/5cb3c452-6196-497a-8d94-3191497ad7f8 
![image](https://user-images.githubusercontent.com/25297859/136384284-16b988e4-5d21-45b7-a755-12a6cf61d5db.png)

cc @luizrrodrigues @Oanh-and-only-Oanh

**Comment by drasticdpk** [comment]
QA pass on prod 
Reference : https://github.com/topcoder-platform/work-manager/issues/1244
